pkg/apis/openebs.io/lvm/v1alpha1/lvmsnapshot.go

```diff
@@ -29,8 +29,27 @@ type LVMSnapshot struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   VolumeInfo `json:"spec"`
-	Status SnapStatus `json:"status"`
+	Spec   LVMSnapshotSpec `json:"spec"`
+	Status SnapStatus      `json:"status"`
+}
+
+// LVMSnapshotSpec defines LVMSnapshot spec
+type LVMSnapshotSpec struct {
+	// OwnerNodeID is the Node ID where the snapshot group is present which is where
+	// the snapshot has been provisioned.
+	// OwnerNodeID can not be edited after the snapshot has been provisioned.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	OwnerNodeID string `json:"ownerNodeID"`
+
+	// VolGroup specifies the name of the snapshot group where the snapshot has been created.
+	// +kubebuilder:validation:Required
+	VolGroup string `json:"volGroup"`
+
+	// Size of the snapshot
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	SnapSize string `json:"snapSize"`
```

Comment:
let us add omitempty here, they is another requirement where snapSize can be empty.